### PR TITLE
enhancement/662

### LINKED ISSUE #662: Arrivals on STARs with no altitude restrictions never descend
Arrival aircraft on STARs that do not have any altitude restrictions cannot "descend via STAR" as the arrival spawn logic assumes. This forces airport designers to either spawn aircraft at or below the airspace ceiling, or have aircraft that can't be controlled.

Therefore, the MCP initialization logic should be adjusted to set the aircraft on a manual (`ALT HOLD`) descent to the airspace ceiling if they are spawned above the airspace ceiling.

### PR DESCRIPTION
**Pull request by erikquinn**
Resolves #662.
Unblocks #560 and #648.

Force descent of arrivals unable to descend via their STAR.

### CODE REVIEW

File: src/assets/scripts/client/aircraft/AircraftController.js [hunk 1]
```diff
@@ -636,9 +636,12 @@ export default class AircraftController {
         // TODO: this may need to be reworked.
         // if we are building a preSpawn aircraft, cap the altitude at 18000 so aircraft that spawn closer to
         // airspace can safely enter controlled airspace properly
-        const altitude = isPreSpawn && spawnPatternModel.category === FLIGHT_CATEGORY.ARRIVAL
-            ? 18000
-            : spawnPatternModel.altitude;
+        let altitude = spawnPatternModel.altitude;
+
+        if (isPreSpawn && spawnPatternModel.category === FLIGHT_CATEGORY.ARRIVAL) {
```
Comment (n8rzz) [inline src/assets/scripts/client/aircraft/AircraftController.js]: very nice 👍

**Comment by erikquinn** [comment]
@n8rzz Tests added!

File: src/assets/scripts/client/aircraft/ModeControl/ModeController.js [hunk 2]
```diff
@@ -359,19 +359,29 @@ export default class ModeController {
...
-        // if applicable, or otherwise to the spawn altitude.
+    initializeForAirborneFlight(bottomAltitude, airspaceCeiling, currentAltitude, currentHeading, currentSpeed) {
         this.setAltitudeFieldValue(bottomAltitude);
-        this.setHeadingFieldValue(currentHeading);
-        this.setSpeedFieldValue(currentSpeed);
         this.setAltitudeVnav();
+
+        // if unable to descend via STAR, force a descent to the top of our airspace
+        if (bottomAltitude === Infinity) {
+            this.setAltitudeFieldValue(Math.min(currentAltitude, airspaceCeiling));
```
Comment (n8rzz) [inline src/assets/scripts/client/aircraft/ModeControl/ModeController.js]: this should be set to a const to provide additional context:
```javascript
const impliedBottomAltitude = Math.min(currentAltitude, airspaceCeiling);
```

or something along those lines

Comment (erikquinn) [reply src/assets/scripts/client/aircraft/ModeControl/ModeController.js]: Ah, of course!

Comment (erikquinn) [reply src/assets/scripts/client/aircraft/ModeControl/ModeController.js]: done

**Pull request merged**